pennylane/devices/qubit/simulate.py

```diff
@@ -112,6 +209,14 @@ def get_final_state(circuit, debugger=None, interface=None):
     for op in circuit.operations[bool(prep) :]:
         state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger)
 
+        # Handle postselection on mid-circuit measurements
+        if isinstance(op, qml.Projector):
+            state, circuit._shots, state_nan = _postselection_postprocess(
+                state, is_state_batched, circuit.shots
+            )
+            if state_nan:
```

Comment:
instead of returning `state_nan`, could we just do a quick check like `np.isnan(nan_state[(0,) * nan_state.ndim])`?